### PR TITLE
Ana/add no mainnet for this address check to signin

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Something important I had forgotten!

"Domesticate" the SessionSignIn too, like it was done with SessionExplore 😉 

This is a work in progress.

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
> # [Codecov](https://codecov.io/gh/luniehq/lunie/pull/3553?src=pr&el=h1) Report
> > Merging [#3553](https://codecov.io/gh/luniehq/lunie/pull/3553?src=pr&el=desc) into [develop](https://codecov.io/gh/luniehq/lunie/commit/fd6599c10ac3c231f59e60e6530e23692ac095cd?src=pr&el=desc) will **decrease** coverage by `0.04%`.
> > The diff coverage is `75%`.
> 
> ```diff
> @@             Coverage Diff             @@
> ##           develop    #3553      +/-   ##
> ===========================================
> - Coverage    88.43%   88.39%   -0.05%     
> ===========================================
>   Files          122      122              
>   Lines         2576     2584       +8     
>   Branches       356      375      +19     
> ===========================================
> + Hits          2278     2284       +6     
> - Misses         287      289       +2     
>   Partials        11       11
> ```
> 
> [Impacted Files](https://codecov.io/gh/luniehq/lunie/pull/3553?src=pr&el=tree)	Coverage Δ	
> [src/components/common/TmSessionSignIn.vue](https://codecov.io/gh/luniehq/lunie/pull/3553/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvY29tbW9uL1RtU2Vzc2lvblNpZ25Jbi52dWU=)	`96.29% <75%> (-3.71%)`	⬇️

That's why I said "work in progress" 😆

File: changes/ana_add-no-mainnet-for-this-address-found-to-sign-in [hunk 1]
```diff
@@ -0,0 +1 @@
+[Changed] [#3553](https://github.com/cosmos/lunie/pull/3553) Adds check in SessionSignIn to verify the signin address belongs to an existing mainnet. Otherwise displays an error message @Bitcoinera
```
Comment (jbibla) [inline changes/ana_add-no-mainnet-for-this-address-found-to-sign-in]: how can we tell if it belongs to a mainnet?

Comment (Bitcoinera) [reply changes/ana_add-no-mainnet-for-this-address-found-to-sign-in]: I mean, maybe I should rephrase this. 

Right now we don't have in place a way to tell whether the account was created in mainnet or in testnet (which I don't think is needed), so there is no way to tell this.

What I wanted to mean here is, it checks if there is an existing mainnet for the network family the address belongs to. If there isn't any (like nowadays in Regen ,for instance), it displays the error message. You need to check the testnet checkbox.

Comment (jbibla) [reply changes/ana_add-no-mainnet-for-this-address-found-to-sign-in]: > You need to check the testnet checkbox.

could we "check it" automatically?

**Comment by jbibla** [comment]
still WIP?

I think this looks good but will wait for someone else's review because i'm not certain about the existing network check logics.

**Comment by Bitcoinera** [comment]
> still WIP?
> 
> I think this looks good but will wait for someone else's review because i'm not certain about the existing network check logics.

sorry 😆 Multidenom rewards got in the way.

**Comment by jbibla** [comment]
code cov needs a bit of love!

**Comment by Bitcoinera** [comment]
> code cov needs a bit of love!

But I am finishing the rewards thing + reviewing & helping in Aleksei's PR 😆 

Also the multi denom ActionModal I need to check everything is good.

This is the next thing after those other tasks.

File: changes/ana_add-no-mainnet-for-this-address-found-to-sign-in (hunk 1 above)
Comment (Bitcoinera) [reply changes/ana_add-no-mainnet-for-this-address-found-to-sign-in]: Actually this is how it should be and it is something I have noticed in our flow which should be fixed

Comment (Bitcoinera) [reply changes/ana_add-no-mainnet-for-this-address-found-to-sign-in]: Oh, yes. But @faboweb said it was better to let users clearly know when they are entering a testnet. Maybe we could just drop a brief warning instead throw a validation error?

Comment (jbibla) [reply changes/ana_add-no-mainnet-for-this-address-found-to-sign-in]: i'm talking about the situation where a user selects a network (gaia testnet, for eg.), then clicks sign in, then enters a cosmos address and ends up on mainnet. if there is a testnet selected, the checkbox should be selected too.

**Comment by jbibla** [comment]
still WIP? e2e actually failing or just needs to be re-run?

**Comment by Bitcoinera** [comment]
> still WIP? e2e actually failing or just needs to be re-run?

Oops sorry. Sometimes I just forget removing it 😅

**Pull request merged**